framework/src/media/MediaRecorderImpl.cpp

```diff
@@ -671,6 +679,11 @@ void MediaRecorderImpl::notifyObserver(recorder_observer_command_t cmd, ...)
 			std::shared_ptr<unsigned char> autodata(data, [](unsigned char *p){ delete[] p; });
 			row.enQueue(&MediaRecorderObserverInterface::onRecordBufferDataReached, mRecorderObserver, mRecorder, autodata, size);
 		} break;
+		case RECORDER_OBSERVER_COMMAND_FLUSH: {
+			medvdbg("RECORDER_OBSERVER_COMMAND_FLUSH\n");
```

Comment:
We can take the mObMtx here.

Reply:
638 line already take.
do you want move?

Reply:
Yes. mObMtx is needed only this case, isn't it?

Reply:
you right,